### PR TITLE
New Berries

### PR DESCRIPTION
**Pull request by cosmiccoincidence**
#### Summary
```SUMMARY: Content "Adds additional berries"```

#### Purpose of change

- [x] Added 4 new fruits (mulberry, elderberry, huckleberry, & rose hips) as well as bushes/trees for them, seeds and seed recipes. Also added bushes for the in-game raspberries, blackberries, and grapes (did not add a bush for cranberry due to shinino adding it in PR #25270).

- [x] Added new fruit to cooking components.

> ~Step 3:~ Add locations that include the berry plants.

Instead I added the plants to regional_map_settings for now, and will add the plants to locations in a different PR due to my other open PR (#25344) added some plants too. Makes sense Ill do all the location stuff together.

### CODE REVIEW

**Comment by cosmiccoincidence** [comment]
I was thinking of adding a thorn bush to the game but instead decided to add thorns to appropriate bushes. I added "sharp" flags to blackberry and raspberry bushes, and in doing so added descriptions to all current berry bushes.

**Comment by cosmiccoincidence** [comment]
The 2 berry trees I personally can't add to mapgen cause it's not in the jsons.

**Comment by cosmiccoincidence** [comment]
Added rose hips as fruit but adding the harvest in #25344

File: data/json/regional_map_settings.json [hunk 1]
```diff
@@ -24,6 +28,10 @@
           "boosted_other": {                "//": "...and the first non-default picked from this list...",
               "t_shrub_blueberry": 40.0,
               "t_shrub_strawberry": 30.0,
+              "t_shrub_blackberry": 35.0,
+              "t_shrub_raspberry": 40.0,
+              "t_shrub_huckleberry": 35.0,
+              "t_shrub_grape": 15,
```
Comment (ZhilkinSerg) [inline data/json/regional_map_settings.json]: image

File: data/json/recipes/food/seed.json [hunk 2]
```diff
@@ -137,6 +137,66 @@
...
+    "subcategory": "CSC_FOOD_SEEDS",
+    "skill_used": "survival",
+    "difficulty": 2,
+    "time": 5000,
+    "autolearn": true,
+    "qualities": [ { "id": "CUT", "level": 1 } ],
+    "components": [ [ [ "huckleberries", 1 ] ] ]
+  },
+  {
+    "result": "seed_grapes",
```
Comment (ZhilkinSerg) [inline data/json/recipes/food/seed.json]: image

File: data/json/items/comestibles.json [hunk 3]
```diff
@@ -4001,6 +4001,196 @@
...
+    "name": "irradiated elderberry",
+    "name_plural": "irradiated elderberries",
+    "description": "An irradiated elderberry will remain edible nearly forever.  Sterilized using radiation, so it's safe to eat.",
+    "proportional": { "price": 2 },
+    "spoils_in": 0,
+    "container": "bag_plastic"
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "rose_hips",
```
Comment (ZhilkinSerg) [inline data/json/items/comestibles.json]: image

File: data/json/terrain.json [hunk 4]
```diff
@@ -3365,6 +3365,108 @@
...
+    "move_cost": 0,
+    "flags": [ "FLAMMABLE_ASH", "NOITEM", "SUPPORTS_ROOF", "TREE", "REDUCE_SCENT" ],
+    "transforms_into": "t_tree_elderburry_harvested",
+    "examine_action": "harvest_ter_nectar",
+    "harvest_by_season": [
+      {
+        "seasons": [ "summer" ],
+        "entries": [
+          { "drop": "elderburries", "base_num": [ 8, 20 ], "scaled_num": [ 0, 0.5 ] },
+          { "drop": "seed_elderburries", "base_num": [ 3, 5 ], "scaled_num": [ 0, 0.25 ] }
```
Comment (ZhilkinSerg) [inline data/json/terrain.json]: image

File: data/json/terrain.json [hunk 5]
```diff
@@ -3836,6 +3940,202 @@
...
+    "move_cost": 8,
+    "flags": [ "TRANSPARENT", "CONTAINER", "FLAMMABLE_ASH", "THIN_OBSTACLE", "SHRUB", "SHORT" ],
+    "transforms_into": "t_shrub_grape_harvested",
+    "examine_action": "harvest_ter_nectar",
+    "harvest_by_season": [
+      {
+        "seasons": [ "summer" ],
+        "entries": [
+          { "drop": "grapes", "base_num": [ 2, 5 ], "scaled_num": [ 0, 0.5 ] },
+          { "drop": "seed_grapes", "base_num": [ 1, 2 ], "scaled_num": [ 0, 0.25 ] }
```
Comment (ZhilkinSerg) [inline data/json/terrain.json]: image

**Comment by cosmiccoincidence** [comment]
Fixes made.

File: data/json/terrain.json [hunk 6]
```diff
@@ -3821,6 +3924,7 @@
     "type": "terrain",
     "id": "t_shrub_strawberry_harvested",
     "name": "strawberry bush",
+    "description": "A small strawberry bush thats fruitless.",
```
Comment (ZhilkinSerg) [inline data/json/terrain.json]: that's

File: data/json/terrain.json [hunk 7]
```diff
@@ -3836,6 +3940,202 @@
...
+      "sound_fail": "brush.",
+      "ter_set": "t_dirt",
+      "items": [ { "item": "withered", "prob": 50, "count": [ 1, 2 ] } ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_shrub_blackberry_harvested",
+    "name": "blackberry bush",
+    "description": "A small blackberry bush thats fruitless.  Watch out for its thorns!",
```
Comment (ZhilkinSerg) [inline data/json/terrain.json]: that's

File: data/json/terrain.json [hunk 8]
```diff
@@ -3836,6 +3940,202 @@
...
+      "sound_fail": "brush.",
+      "ter_set": "t_dirt",
+      "items": [ { "item": "withered", "prob": 50, "count": [ 1, 2 ] } ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_shrub_huckleberry_harvested",
+    "name": "huckleberry bush",
+    "description": "A small huckleberry bush thats fruitless.",
```
Comment (ZhilkinSerg) [inline data/json/terrain.json]: that's

File: data/json/terrain.json [hunk 9]
```diff
@@ -3836,6 +3940,202 @@
...
+      "sound_fail": "brush.",
+      "ter_set": "t_dirt",
+      "items": [ { "item": "withered", "prob": 50, "count": [ 1, 2 ] } ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_shrub_raspberry_harvested",
+    "name": "raspberry bush",
+    "description": "A small raspberry bush thats fruitless.  Watch out for its thorns!",
```
Comment (ZhilkinSerg) [inline data/json/terrain.json]: that's

File: data/json/terrain.json [hunk 10]
```diff
@@ -3836,6 +3940,202 @@
       "items": [ { "item": "withered", "prob": 50, "count": [ 1, 2 ] } ]
     }
   },
+  {
+    "type": "terrain",
+    "id": "t_shrub_blackberry",
+    "name": "blackberry bush",
+    "description": "A small bush of delicious blackberries.  Watch out for its thorns!",
+    "symbol": "#",
+    "color": "light_black_green",
```
Comment (ZhilkinSerg) [inline data/json/terrain.json]: There is no such color - you should probably use `black_green`.

**Comment by ZhilkinSerg** [comment]
Some small typos left - otherwise looks good.

**Pull request merged**